src/ORM/Association/HasMany.php

```diff
@@ -149,7 +149,11 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
         $options['_sourceTable'] = $this->source();
 
         if ($this->_saveStrategy === self::SAVE_REPLACE) {
-            $this->_unlinkAssociated($properties, $entity, $target, $targetEntities);
+            $unlinkSuccessful = $this->_unlinkAssociated($properties, $entity, $target, $targetEntities);
+        }
+
+        if (isset($unlinkSuccessful) && !$unlinkSuccessful) {
```

Comment:
Or don't use a variable at all since it's never used anywhere else. Directly returning `false` depending on `$this->_unlinkAssociated()` should be fine.